test/functorch/test_control_flow.py

```diff
@@ -42,8 +52,83 @@ def f(x, y):
         xs = torch.ones(3, 2, 2, device="cuda")
         y = torch.ones(2, device="cuda")
         res = control_flow.map(f, xs, y)
+        expected = _fake_map(f, xs, y)
+        self.assertEqual(expected, res)
+
+    def test_map_list_in_out(self):
+        def f(x, y):
+            return [[x[0][0] + y]]
 
-        self.assertEqual(res, control_flow.map(f, torch.ones(3, 2, 2), torch.ones(2)))
+        xs = [[torch.ones(3, 2, 2)]]
+        y = torch.ones(2)
+        res = control_flow.map(f, xs, y)
+        expected = _fake_map(f, xs, y)
+        self.assertEqual(len(res), 1)
+        self.assertEqual(len(res[0]), 1)
+        self.assertEqual(expected, res)
+
+    def test_map_dict_in_out(self):
```

Comment:
One recommendation for testing is: we should add a `control_flow_opinfo_db`, like the one we have for autograd.Function: https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/autograd_function_db.py

The idea is that we write up a couple of map examples (maybe nested map) as OpInfos, and then send them through the existing ProxyTensor / AOTAutograd / autograd opinfo tests and check that everything works out.

Reply:
Sure! I'll add them.